components/SerialServer/src/serial.c

```diff
@@ -310,9 +311,21 @@ static void internal_putchar(int b, int c)
     error = serial_unlock();
 }
 
+static int getchar_index_from_badge(int badge)
+{
+    for (int i = 0; i < num_getchar_clients; i++) {
+        if (getchar_clients[i].client_id == badge) {
+            return getchar_clients[i].idx;
+        }
+    }
+    return -1;
+}
+
 static void internal_raw_putchar(int id, int c)
 {
-    getchar_client_t *client = &getchar_clients[id];
+    int idx = getchar_index_from_badge(id);
+    ZF_LOGF_IF(idx == -1, "Badge %d is not valid\n", id);
```

Comment:
I don't think this case is really fatal. We have the limitation of 10 clients now, so this client should simply be ignored then.
```
if (idx == -1) {
    ZF_LOGW("Client with badge %d is not supported\n", id);
    return
}
```

Reply:
Done.